Cabal/Distribution/Simple/Program/GHC.hs

```diff
@@ -745,8 +745,8 @@ renderGhcOptions comp _platform@(Platform _arch os) opts
   ---------------
   -- Inputs
 
-  , [ prettyShow modu | modu <- flags ghcOptInputModules ]
   , flags ghcOptInputFiles
+  , [ prettyShow modu | modu <- flags ghcOptInputModules ]
```

Comment:
I'm really blid, but I don't see any difference even GitHub is highlighting it to me?

Reply:
The change is in the ordering.  By loading the main module first (the input file) the ghci session has that module in scope, rather than the first module defined in `other-modules`.

So for instance with this change and a test-suite of

```
  main-is:             Main.hs
  other-modules:
      FooSpec
```

then `Main` will be in scope rather than `FooSpec`.

Reply:
I see now. This is so subtle, we need a comment here.

Reply:
Oh, I agree.  I can do a comment but I'd prefer a test the captures the over-arching desire.

Are there any existing tests that make assertions on `cabal ghci`'s behaviour?

Reply:
I'm afraid we don't have any, as testing interactive behaviour is tricky.

Reply:
Hmm, what about testing to the point of asserting how ghc is invoked?  `renderGhcOptions` is just seems too generic removed from "I need ghci to be invoked this way".  But maybe I should just test around that?

Reply:
Hmm, what about testing to the point of asserting how ghc is invoked?  `renderGhcOptions` is just seems too generic removed from "I need ghci to be invoked this way".  But maybe I should just test around that?

Reply:
Generally, having being able to write test "`cabal repl` loads my package" would be great, but it's not a blocker for this PR.